### ISSUE TITLE
Short explanation of kitti_multiloss.py vs fcn.py

### ISSUE THREAD

**Issue by villanuevab** [MarvinTeichmann/KittiSeg#64]
Hello, 

Thank you again for being so responsive and for this great work. I'm trying to understand the training pipeline and it doesn't look like both of `decoder/fcn.py` and `decoder/kitti_multiloss.py` are used, only `kitti_multiloss.py` is. They also seem to be doing very similar things, at least in the `decoder()` function: `fcn.py` does the upsampling explicitly within `decoder()` whereas `kitti_multiloss.py` takes advantage of that the final upsampled logits feature map (i.e., `upscore32`) is already stored in a `logits['fcn_logits']` dict, since it was calculated in `tensorflow-fcn/fcn8_vgg.py`. 

My questions: 
1. Are both required in the vanilla KittiSeg training pipeline? 
2. How are `fcn.py` and `kitti_multiloss.py` functionally different?

**Comment by MarvinTeichmann** [comment]
> My questions: 
> 1. Are both required in the vanilla KittiSeg training pipeline? 

No, they are alternative implementation of the same kind of decoder. Which decoder is to be used is defined in the hypes. If you train with `kittiseg.json` you will be using `../decoder/kitti_multiloss.py` as specified in the `model` section of the config file. 

`  "model": {
    "input_file": "../inputs/kitti_seg_input.py",
    "architecture_file" : "../encoder/fcn8_vgg.py",
    "objective_file" : "../decoder/kitti_multiloss.py",
    "optimizer_file" : "../optimizer/generic_optimizer.py",
    "evaluator_file" : "../evals/kitti_eval.py"
},`

> 2. How are `fcn.py` and `kitti_multiloss.py` functionally different? 

`kitti_multiloss.py` assumes that the architecture file implements a full segmentation model (i.e. is already an fcn implementation). It only adds a loss to the graph. This can be used in combination with `../encoder/fcn8_vgg.py`. `fcn.py` assumes that the encoder implements a classification nentwork. FCN will transform the classification network into a segmentation network using FCNs and add a loss. This is useful if you want to utilize existing network implementations such as resnet.

**Comment by villanuevab** [comment]
Got it. Thank you!
